From 122da690b39a67cd6e6bce4b9ad3ac1470cc1ed5 Mon Sep 17 00:00:00 2001
From: Alex Brachet <abrachet@google.com>
Date: Wed, 8 Jun 2022 16:21:53 +0000
Subject: [PATCH] [libc] Fix build when __FE_DENORM is defined

[nd: drop hunk that doesn't apply to 14.0.1 due to missing
  commit 805899e68a83 ("[libc] Change FEnv to use MXCSR as source of truth")]

Differential revision: https://reviews.llvm.org/D127222
---
 libc/src/__support/FPUtil/x86_64/FEnvImpl.h | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/libc/src/__support/FPUtil/x86_64/FEnvImpl.h b/libc/src/__support/FPUtil/x86_64/FEnvImpl.h
index 846996ec93af..6bac3e9e6a0f 100644
--- a/libc/src/__support/FPUtil/x86_64/FEnvImpl.h
+++ b/libc/src/__support/FPUtil/x86_64/FEnvImpl.h
@@ -73,7 +73,7 @@ static inline uint16_t get_status_value_for_except(int excepts) {
   // bit flags in the control registers.
   return (excepts & FE_INVALID ? ExceptionFlags::INVALID : 0) |
 #ifdef __FE_DENORM
-         (excepts & __FE_DENORM ? ExceptionFlags::Denormal : 0) |
+         (excepts & __FE_DENORM ? ExceptionFlags::DENORMAL : 0) |
 #endif // __FE_DENORM
          (excepts & FE_DIVBYZERO ? ExceptionFlags::DIV_BY_ZERO : 0) |
          (excepts & FE_OVERFLOW ? ExceptionFlags::OVERFLOW : 0) |
@@ -84,7 +84,7 @@ static inline uint16_t get_status_value_for_except(int excepts) {
 static inline int exception_status_to_macro(uint16_t status) {
   return (status & ExceptionFlags::INVALID ? FE_INVALID : 0) |
 #ifdef __FE_DENORM
-         (status & ExceptionFlags::Denormal ? __FE_DENORM : 0) |
+         (status & ExceptionFlags::DENORMAL ? __FE_DENORM : 0) |
 #endif // __FE_DENORM
          (status & ExceptionFlags::DIV_BY_ZERO ? FE_DIVBYZERO : 0) |
          (status & ExceptionFlags::OVERFLOW ? FE_OVERFLOW : 0) |
@@ -273,8 +273,8 @@ static inline int raise_except(int excepts) {
   if (status_value & internal::ExceptionFlags::INEXACT)
     raise_helper(internal::ExceptionFlags::INEXACT);
 #ifdef __FE_DENORM
-  if (statusValue & internal::ExceptionFlags::Denormal) {
-    raiseHelper(internal::ExceptionFlags::Denormal);
+  if (status_value & internal::ExceptionFlags::DENORMAL) {
+    raise_helper(internal::ExceptionFlags::DENORMAL);
   }
 #endif // __FE_DENORM
 
-- 
2.37.0.144.g8ac04bfd2-goog

